File: QE-Selenium-Java/Activity11.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.WebElement;
import java.time.Duration;

public class Activity11 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        Actions actions = new Actions(driver);

        driver.get("https://training-support.net/webelements/dynamic-controls");

        System.out.println("Page Title: " + driver.getTitle());

        WebElement checkbox = driver.findElement(
                By.xpath("//input[@type='checkbox']"));

        WebElement toggleButton = driver.findElement(By.xpath("//button[contains(text(), 'Toggle Checkbox')]"));

        toggleButton.click();

        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.xpath("//input[@type='checkbox']")));

        System.out.println("Checkbox dissappeared");

        toggleButton.click();

        checkbox = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@type='checkbox']")));

        checkbox.click();
        System.out.println("Checkbox selected");

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-controls")

print("Page Title:", driver.title)

checkbox = driver.find_element(By.XPATH, "//input[@type='checkbox']")

toggle_btn = driver.find_element(
    By.XPATH, "//button[contains(text(),'Toggle Checkbox')]"
)

toggle_btn.click()

wait.until(
    EC.invisibility_of_element_located(
        (By.XPATH, "//input[@type='checkbox']")
    )
)
print("Checkbox disappeared")

toggle_btn.click()

checkbox = wait.until(
    EC.presence_of_element_located(
        (By.XPATH, "//input[@type='checkbox']")
    )
)

checkbox.click()
print("Checkbox selected")

driver.quit()
